Emails: Fix empty margin when resizing browser to a mobile settings

AllTerrainDeveloper opened the pull request:
Fix empty margin when resizing browser to mobile settings on domains and emails step

#### Changes proposed in this Pull Request
Add mobile breaks to add a small margin (20px) and remove an extra margin that the input text form had for domains lookup.



#### Testing instructions

1. Check that when resizing the browser at 1280px width, there is a margin on the left side of the emails step.

BEFORE | AFTER
------------ | -------------
![image](https://user-images.githubusercontent.com/5689927/128041707-49627d58-4b02-4734-ad9d-1c77f6276e10.png) | ![image](https://user-images.githubusercontent.com/5689927/128041774-6d012a90-817c-4eae-81ac-aed9cad5fd4a.png)

2. Check that when resizing the browser at 1080px width, there is a margin on the left side on the domains step.

BEFORE | AFTER
------------ | -------------
![image](https://user-images.githubusercontent.com/5689927/128042004-50230e12-01a0-4389-bdd6-1e0efd08d322.png) | ![image](https://user-images.githubusercontent.com/5689927/128042212-ed951ffc-f29c-4fc8-9833-f484688ff96f.png)

Related to {1200182182542585-as-1200703026939303}

AllTerrainDeveloper committed:
Emails: Fix empty margin when resizing browser to a mobile settings on domains and emails step

stephanethomas reviewed (changes requested):
I'm getting this weird display issue depending on the width of the browser for the domain step:

<img width="445" alt="screenshot" src="https://user-images.githubusercontent.com/594356/128185519-e9ea5c94-06ed-4bac-8886-2d82dad31a09.png">

Note that if fixing that particular page is becoming too difficult, I'm totally fine with reverting our fix and letting Nomado fix it.

Regarding the email step, it no longer has any issue with the left margin but I've noticed it doesn't display well on small screens:

<img width="250" alt="screenshot" src="https://user-images.githubusercontent.com/594356/128185977-79a466ba-cbb6-4408-9aab-9a24a92f2536.png">

I don't think that was caused by your changes but that's something we should address. I suspect we have mockups for mobile devices but if not maybe we should discuss that issue with @saygunnyc.

saygunnyc commented:
Thanks for the ping @stephanethomas !
Yes, we have mobile designs. I just checked this over an iPhone 11, Chrome Browser (up to date), and the mobile version renders as designed.

![iPhone](https://user-images.githubusercontent.com/78876323/128261560-19cdf864-2e11-4867-a8c0-7ddae21efebf.PNG)

However, when I check over the desktop and scale down the browser, this is what happens.

<img width="500" alt="Screen Shot 2021-08-04 at 6 07 46 PM" src="https://user-images.githubusercontent.com/78876323/128261654-6c72f54d-1770-40da-92bc-17b90cfd9990.png">

AllTerrainDeveloper commented:
> I'm getting this weird display issue depending on the width of the browser for the domain step:
> 
> <img alt="screenshot" width="445" src="https://user-images.githubusercontent.com/594356/128185519-e9ea5c94-06ed-4bac-8886-2d82dad31a09.png">
> 
> Note that if fixing that particular page is becoming too difficult, I'm totally fine with reverting our fix and letting Nomado fix it.
> 
> Regarding the email step, it no longer has any issue with the left margin but I've noticed it doesn't display well on small screens:
> 
> <img alt="screenshot" width="250" src="https://user-images.githubusercontent.com/594356/128185977-79a466ba-cbb6-4408-9aab-9a24a92f2536.png">
> 
> I don't think that was caused by your changes but that's something we should address. I suspect we have mockups for mobile devices but if not maybe we should discuss that issue with @saygunnyc.

I can't replicate the first issue you are having, like having a big gap between input search and suggestions. I'll keep trying. but I'll focus on mobile view too.

stephanethomas commented:
> I can't replicate the first issue you are having, like having a big gap between input search and suggestions.

FWIW, I was using Chrome latest on Windows 10.

AllTerrainDeveloper committed:
Emails: Sign up media breakpoints for smaller devices.

AllTerrainDeveloper commented:
I've reworked a bit the CSS styling, i added changes for emails but I have added anything to domains.

Emails:
![resize](https://user-images.githubusercontent.com/5689927/128360839-00663186-504f-495d-9e22-8def4992891f.gif)

Domains:
![resize domains](https://user-images.githubusercontent.com/5689927/128361508-017ae646-7dd8-402b-9316-0e200fa5d795.gif)

daledupreez commented on client/signup/style.scss:
```diff
@@ -712,6 +712,14 @@ body.is-section-signup.is-white-signup .layout:not( .dops ):not( .is-wccom-oauth
 	.signup__step.is-domains,
 	.signup__step.is-emails {
+		@include break-large {
+			margin: 0 0 0 20px;
+			padding-left: 20px;
```
We're getting 40px on the left here -- is there a reason you're using both margin and padding here? If we really need the left padding, it may make sense when we shift to `break-wide` below when we move back to `margin: initial`.

AllTerrainDeveloper replied:
I've addressed it :) Plus added a few more fixes in domains.

daledupreez reviewed (commented):
Wow... the CSS here is really non-ideal, as the left padding/margin/spacing is all over the place across the various elements, with padding getting added for each element, and not always at the same breakpoints. 😬 

With the one exception of the double-left padding, I think we can ship this, but I think it would be good to queue up a broader review of the CSS.

AllTerrainDeveloper committed:
Emails: Pull request suggestions plus additional fixes

AllTerrainDeveloper commented:
Within the last commit, I've also included a new fix for domain margin. All should be fine now.
![resize domains](https://user-images.githubusercontent.com/5689927/128700969-c26a1c1f-af8f-4f0d-8e6e-f5923605e15b.gif)

daledupreez reviewed (approved):
LGTM! 🚢
